hack/api-migration/main.go

```diff
@@ -0,0 +1,203 @@
+
+		"certmanager.k8s.io/issuer":           "cert-manager.io/issuer",
+		"certmanager.k8s.io/cluster-issuer":   "cert-manager.io/cluster-issuer",
+		"certmanager.k8s.io/alt-names":        "cert-manager.io/alt-names",
+		"certmanager.k8s.io/ip-sans":          "cert-manager.io/ip-sans",
+		"certmanager.k8s.io/common-name":      "cert-manager.io/common-name",
+		"certmanager.k8s.io/issuer-name":      "cert-manager.io/issuer-name",
+		"certmanager.k8s.io/issuer-kind":      "cert-manager.io/issuer-kind",
+		"certmanager.k8s.io/certificate-name": "cert-manager.io/certificate-name",
+	}
+
+	depreciations = []string{
+		"certmanager.k8s.io/acme-challenge-type",
+		"certmanager.k8s.io/acme-dns01-provider",
+	}
+)
+
+var cmd = &cobra.Command{
+	Use:  "cert-manager-api-migration",
+	Long: "A tool to download Ingress resources and convert cert-manager annotations from certmanager.k8s.io to cert-manager.io. This tool will not modify resources in the API server.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(kubeconfig) == 0 {
+			kubeconfig = os.Getenv("KUBECONFIG")
+		}
+
+		if len(kubeconfig) == 0 {
+			return errors.New("flag --kubeconfig or environment variable $KUBECONFIG must be set")
+		}
+
+		client, err := clientSet(kubeconfig)
+		if err != nil {
+			return err
+		}
+
+		ingList, err := client.NetworkingV1beta1().Ingresses("").List(metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		ingStr, err := writeIngressToFile(ingList, origFile)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("written current ingresses to file %q\n", origFile)
+		fmt.Printf("searching ingress resources for occurrences of old API annotations...\n")
+
+		for oldA, newA := range migrations {
+			count := strings.Count(ingStr, oldA)
+			if count == 0 {
+				continue
+			}
+
+			fmt.Printf("found %d instances of %q\tmigrating to %q\n", count, oldA, newA)
+
+			ingStr = strings.ReplaceAll(ingStr, oldA, newA)
+		}
+
+		for _, d := range depreciations {
+			count := strings.Count(ingStr, d)
+			if count == 0 {
+				continue
+			}
+
+			fmt.Printf("found %d instances of %q\tthis field is DEPRECIATED and will be deleted\n", count, d)
+
+			lines := strings.Split(ingStr, "\n")
+
+			for i, line := range lines {
+				if strings.Contains(line, d) {
+					lines = append(lines[:i], lines[i+1:]...)
+				}
+			}
+
+			ingStr = strings.Join(lines, "\n")
+		}
+
+		count := strings.Count(ingStr, oldGroupName)
+		if count > 0 {
+			fmt.Fprintf(os.Stderr,
+				"found %d more instances of the group %q with unrecognised paths\n", count, oldGroupName)
+		}
+
+		if err := ioutil.WriteFile(newFile, []byte(ingStr), 0644); err != nil {
+			return err
+		}
+
+		fmt.Printf("written new ingresses to file %q\n", newFile)
+		fmt.Printf("\nPlease check for any missing or wrong annotations in your newly generated ingress manifests.\n")
+		fmt.Printf("You are encouraged to check the diff of the two files to determine what has changed (diff %s %s).\n",
```

Comment:
Can you make this last message output like:

```
You should now check the diff between the two files to determine what has changed, like so:

    diff %s %s
```

and/or perhaps run this command for them? Else I bet a load of people won't look 😄

Reply:
Maybe write the diff to a file otherwise the output can get quite noisy?

`diffPath := oldPath + "." newPath + ".diff"` ?